src/main/java/no/fintlabs/azure/storage/StorageResourceRepository.java

```diff
@@ -59,18 +62,44 @@ public void update(StorageResource storageResource) {
         storageResources.put(storageResource.getStorageAccountName(), storageResource);
     }
 
-    public long size() {
-        return storageResources.size();
+    public long size(String environment) {
+        return storageResources.values().stream()
+                .filter(storageResource -> storageResource.getEnvironment().equals(environment))
+                .count();
     }
 
-    private void loadStorageResources() {
+    public void refresh(String environment) {
+        Collection<StorageResource> storageResources = getStorageResourcesByEnvironment(environment);
+        storageResources.forEach(storageResource -> {
+            loadStorageResources();
+        });
+    }
```

Comment:
Logikken her henger ikke helt sammen. Først hentes alle eksisterende ressurser, og deretter itereres det over dem for å hente alle ressursene på nytt for hver enkelt ressurs. For eksempel, hvis du har 5 ressurser i cache, vil du hente alle ressursene på nytt 5 ganger – én gang for hver ressurs.

Dette trenger kun å være  `loadStorageResources()`